src/emulator/storage/apis/firebase.ts

```diff
@@ -342,9 +347,14 @@ export function createFirebaseEndpoints(emulator: StorageEmulator): Router {
         upload = uploadService.finalizeResumableUpload(uploadId);
       } catch (err) {
         if (err instanceof NotFoundError) {
+          uploadService.setResponseCode(uploadId, 404);
```

Comment:
do you not need to update the GCS API?

Reply:
That's fair, I more meant like what if you were to send direct HTTP requests to `finalize` via `supertest`, but we can say that's out of scope.